### ISSUE TITLE
No hover effect on non-flat buttons

### ISSUE THREAD

**Issue by adrianmc** [react-toolbox/react-toolbox#621]
The issue is pretty straight-forward. The current release of React-Toolbox has Buttons that only show a hover state if they are flat. You can try it [here](http://react-toolbox.com/#/components/button) in the docs.

All of the other major competitors have a hover state for raised buttons:

- [Material UI](http://www.material-ui.com/#/components/raised-button)
- [Materialize](http://materializecss.com/buttons.html)

Also, the [official spec](https://material.google.com/components/buttons.html#buttons-raised-buttons) from Google has a hover state:

[![Official Spec](https://material-design.storage.googleapis.com/publish/material_v_8/material_ext_publish/0B8olV15J7abPZm1EakJMZWxSdXc/components_buttons_main17.png)](https://material.google.com/components/buttons.html#buttons-raised-buttons)

The hover state is pretty important for UX, and it doesn't make sense to restrict it only to flat buttons. 

Places in the code where this restriction to `.flat` buttons can be seen:

- Lines `146` and `161` of [components/button/theme.scss](https://github.com/react-toolbox/react-toolbox/blob/6cd4202c6cde14f404b923a0817566b3520dc5a2/components/button/theme.scss)

- Line `13` of [components/button/_mixins.scss](https://github.com/react-toolbox/react-toolbox/blob/d7f8e202d23bd272defe40e1b3638ae8dcd5d041/components/button/_mixins.scss)

**Comment by javivelasco** [comment]
Although I get your point, I don't fully agree. As you can see in the official spec it's made a difference between desktop and mobile. The screenshot attached implements a raised button with no elevation so it get's the elevation on hover. By default we are implementing raised button as an elevated piece of material by default meaning the only way to hover would be to elevate the button even more. This is violating the spec though so probably a good solution would be to provide a variant for desktop with no elevation and another one with elevation for whoever wants to use it

**Comment by javivelasco** [comment]
In fact, if you check what in my opinion is the most accurate implementation of MD [here](https://getmdl.io/components/index.html#buttons-section) you'd see that raised buttons has no hover state...

**Comment by adrianmc** [comment]
What about lowering the opacity on the background color like Material-UI? I think having some kind of hover state is better than nothing at all, wouldn't you agree? Or at least give an option to do it, but not make it the default. @javivelasco

**Comment by javivelasco** [comment]
Yeah, I agree with you that some kind of effect on hover is good UI feedback and is probably needed. I also would like to be as close as possible to the spec so, since this is desktop, I think the best approach would be to identify the platform and display one way of another

**Comment by adrianmc** [comment]
@javivelasco do you have any suggestions on how to proceed? I can make an attempt at a PR if it is not too difficult.
